### PR TITLE
charm logging: add abspath

### PR DESCRIPTION
**Pull request by nammn**
- adds abspath output, as is could be relative, which makes it hard for debugging purposes
- some unit tests with it

### CODE REVIEW

File: charmdir.go [hunk 1]
```diff
@@ -534,6 +534,15 @@ func (dir *CharmDir) MaybeGenerateVersionString(logger Logger) (string, string,
 	vcsOrder := [...]string{"git", "hg", "bzr"}
 	cmdWaitTime := 2 * time.Second
 
+	absPath := dir.Path
+	if !filepath.IsAbs(absPath) {
+		var err error
+		absPath, err = filepath.Abs(dir.Path)
+		if err != nil {
+			logger.Warningf("failed resolving relative path %q", dir.Path)
```
Comment (nammn) [inline charmdir.go]: We don't want to fail here, as this would change the long-running behavior we had until now imo

Comment (achilleasa) [reply charmdir.go]: So, what happens if this fails and you fall through? Won't it cause issues with the code below? I think that absPath will be set to `""`

Comment (nammn) [reply charmdir.go]: Hmm you are right, i suppose fallback to path would be the way to have no regression?

Comment (nammn) [reply charmdir.go]: returning the error now.

**Comment by nammn** [comment]
$$merge$$ thanks for the rev!

**Pull request merged**